algo/src/test/java/com/hartwig/serve/sources/ckb/ActionableTrialFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+package com.hartwig.serve.sources.ckb;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.Lists;
+import com.hartwig.serve.ckb.datamodel.CkbEntry;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ClinicalTrial;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ImmutableLocation;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ImmutableVariantRequirementDetail;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.VariantRequirementDetail;
+import com.hartwig.serve.datamodel.EvidenceDirection;
+import com.hartwig.serve.datamodel.EvidenceLevel;
+import com.hartwig.serve.datamodel.Knowledgebase;
+
+import org.apache.commons.compress.utils.Sets;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+public class ActionableTrialFactoryTest {
+
+    @Test
+    public void canCreateActionableTrial() {
+        Location location = ImmutableLocation.builder().nctId("nctid").city("").country("Netherlands").status("Recruiting").build();
+        VariantRequirementDetail requirementType =
+                ImmutableVariantRequirementDetail.builder().profileId(0).requirementType("required").build();
+        CkbEntry entry = CkbTestFactory.createEntryWithClinicalTrial(location, "Recruiting", requirementType);
+        ActionableTrialFactory actionableTrialFactory = new ActionableTrialFactory();
+        Set<ActionableEntry> entrySet = actionableTrialFactory.create(entry, "KRAS", "gene");
+
+        assertEquals(1, entrySet.size());
+        ActionableEntry trial = entrySet.iterator().next();
+        assertEquals(Knowledgebase.CKB_TRIAL, trial.source());
+        assertEquals("KRAS", trial.sourceEvent());
+        assertEquals(Sets.newHashSet("https://clinicaltrials.gov/study/nctid"), trial.sourceUrls());
+        assertEquals("title", trial.treatment().name());
+        assertEquals("AB", trial.applicableCancerType().name());
+        assertEquals("162", trial.applicableCancerType().doid());
+        assertEquals(Sets.newHashSet(), trial.blacklistCancerTypes());
+        assertEquals(EvidenceLevel.B, trial.level());
+        assertEquals(EvidenceDirection.RESPONSIVE, trial.direction());
+        assertEquals(Sets.newHashSet("Netherlands"), trial.evidenceUrls());
```

Comment:
This test contains a fair amount of dependencies on CkbTestFactory which is unwanted. If you don't explicitly set a value when creating the CkbEntry you also shouldn't test for it afterwards. E.g. nowhere you set the applicable cancer type name to "AB" yet you test for it.